src/common/reducers/snaps.js

```diff
@@ -39,14 +42,17 @@ export function snaps(state = {
         success: false,
         error: null
       };
+      // XXX a little confusing because we're not refactoring this yet, just
+      // making do for the repositories refactor
     case ActionTypes.FETCH_SNAPS_SUCCESS:
       return {
         ...state,
         isFetching: false,
         success: true,
         snaps: [
-          ...action.payload
+          ...action.payload.payload.snaps
         ],
+        ids: union(state.ids, action.payload.result),
```

Comment:
Adding this along side the current implementation, with the goal of moving to similar pattern for snaps as established here in repositories.